x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EsqlActionIT.java

```diff
@@ -861,6 +861,7 @@ public void testFromStatsLimit() {
         }
     }
 
+    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/99826")
     public void testFromLimit() {
         try (EsqlQueryResponse results = run("from test | keep data | limit 2")) {
```

Comment:
There's something funky going on with `LimitOperator` and I tried to track it down - you can see the tests I added. But I ultimately gave up. I think the problem here is that we `finish` the `Operator` and something on the exchange doesn't close. I'm not sure though.